tests/unit/style/parsing/effects.rs

```diff
@@ -33,3 +34,23 @@ fn test_clip() {
                                    "rect(auto, auto, auto, auto)");
 }
 
+#[test]
+fn test_longhands_parse_origin() {
+    let url = ServoUrl::parse("http://localhost").unwrap();
+    let context = ParserContext::new(Origin::Author, &url, Box::new(CSSErrorReporterTest));
+
+    let mut parser = Parser::new("1px 2px rubbish");
+    let parsed = longhands::parse_origin(&context, &mut parser);
+    assert_eq!(parsed.is_ok(), true);
```

Comment:
Actually, not in this case. Exhaustion is checked in another part of the code, not in longhand parsing. This parser will return `Ok` but fail at `is_exhausted`.

And there is `assert_parser_exhausted` macro for this kind of tests. Could you use this one please? You can see the usage in `tests/unit/style/parsing/border.rs`.